dataprofiler/tests/speed_tests/structured_throughput_testing.py

```diff
@@ -0,0 +1,114 @@
+import time
+import json
+import random
+from collections import defaultdict
+
+import numpy as np
+import tensorflow as tf
+
+try:
+    import sys
+    sys.path.insert(0, '..')
+    import dataprofiler as dp
+except ImportError:
+    import dataprofiler as dp
+    
+
+# suppress TF warnings
+tf.compat.v1.logging.set_verbosity(tf.compat.v1.logging.ERROR)
+
+
+################################################################################
+######################## set any optional changes here #########################
+################################################################################
+options = dp.ProfilerOptions()
+
+# these two options default to True if commented out
+options.structured_options.multiprocess.is_enabled = False
+# options.structured_options.data_labeler.is_enabled = False
+
+# parameter alteration
+ALLOW_SUBSAMPLING = True # profiler to subsample the dataset if large
+PERCENT_TO_NAN = 0.0  # Value must be between 0 and 100
+
+
+sample_sizes = [100, 1000, 5000, 7500, int(1e5)]
+################################################################################
+
+if __name__ == "__main__":
+
+    # set seed
+    random.seed(0)
+    np.random.seed(0)
+    dp.set_seed(0)
+
+
+    # load data
+    data = dp.Data('data/time_structured_profiler.csv')
+
+    # [0] allows model to be initialzied and added to labeler
+    sample_sizes = [0] + sample_sizes
+    profile_times = []
+    for sample_size in sample_sizes:
+        # setup time dict
+        
+        print(f"Evaluating sample size: {sample_size}")
+        df = data.data.sample(sample_size, replace=True).reset_index(drop=True)
+        
+        if PERCENT_TO_NAN:
+            samples_to_nan = int(len(df) * PERCENT_TO_NAN / 100)
+            for col_name in df:
+                ind_to_nan = random.sample(list(df.index), samples_to_nan)
+                df[col_name][ind_to_nan] = 'None'
```

Comment:
can we also add None or np.nan as well?

Reply:
The intent is purely to test throughput, I personally didn't feel like adding these would have the impact as we don't handle them differently. Happy to chat more on it though.